rlkit/samplers/data_collector/base_collector.py

```diff
@@ -0,0 +1,43 @@
+import abc
+
+class BaseCollector(object, metaclass=abc.ABCMeta):
+    def end_epoch(self, epoch):
+        pass
+
+    def get_diagnostics(self):
+        return {}
+
+    def get_snapshot(self):
+        return {}
+
+    @abc.abstractmethod
+    def get_epoch_paths(self):
+        pass
+
+class PathCollector(BaseCollector, metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def collect_new_paths(
+            self,
+            max_path_length,
+            num_steps,
+            discard_incomplete_paths,
+    ):
+        pass
+
+class StepCollector(BaseCollector, metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def collect_new_step(
+            self,
+            max_path_length,
+            num_steps,
+            discard_incomplete_paths,
+    ):
+        pass
+
+    @abc.abstractmethod
+    def start_collection(self):
+        pass
+
+    @abc.abstractmethod
+    def end_collection(self):
```

Comment:
Are start/end necessary given end_epoch and get_epoch_paths?

Reply:
I think start_collection is kind of necessary to call environment.reset(). Though this can be done by checking if the observation cache .obs is set to None or not, if None call .reset(). And then it needs to clear this cache in the end_epoch() method. Would you like this more?